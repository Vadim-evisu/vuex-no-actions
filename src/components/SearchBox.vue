<template>
  <div>
    <form action="" v-on:submit="triggerSearch">
      <input type="text" id="searchBox" v-model="searchBoxText" class="input" />
      <button id="searchSubmitButton" class="button is-primary">Search</button>
    </form>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
export default Vue.extend({
  data: function () {
    return {
      searchBoxText: '',
    };
  },
  methods: {
    triggerSearch: function (event: Event) {
      event.preventDefault();
      if (!this.searchBoxText) {
        return;
      }
      let encoded = encodeURIComponent(this.searchBoxText);
      this.$router.push({ name: 'Search', query: { s: encoded } });
    },
  },
});
</script>
