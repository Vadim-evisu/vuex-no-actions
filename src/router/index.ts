import Vue from 'vue';
import VueRouter, { RouteConfig } from 'vue-router';
import Home from '../views/Home.vue';
import Search from '../views/Search.vue';
import Person from '../views/Person.vue';
import Movie from '../views/Movie.vue';

Vue.use(VueRouter);

const routes: Array<RouteConfig> = [
  {
    path: '/',
    name: 'Home',
    component: Home,
  },
  {
    path: '/search',
    name: 'Search',
    component: Search,
  },
  {
    path: '/about',
    name: 'About',
    // route level code-splitting
    // this generates a separate chunk (about.[hash].js) for this route
    // which is lazy-loaded when the route is visited.
    component: () =>
      import(/* webpackChunkName: "about" */ '../views/About.vue'),
  },
  {
    path: '/movie',
    name: 'Movie',
    component: Movie,
  },
  {
    path: '/person',
    name: 'Person',
    component: Person,
  },
];

const router = new VueRouter({
  mode: 'history',
  base: process.env.BASE_URL,
  routes,
});

export default router;
