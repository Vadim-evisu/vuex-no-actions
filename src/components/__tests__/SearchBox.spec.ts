import { createLocalVue, mount, shallowMount, Wrapper } from '@vue/test-utils';
import VueRouter from 'vue-router';
import SearchBox from '../SearchBox.vue';
import router from '@/router';

describe('SearchBox tests', () => {
  it('test sanity check', () => {
    expect(true).toBeTruthy();
  });
});

describe('SearchBox', () => {
  const wrapper = shallowMount(SearchBox);
  it('matches the snapshot when mounted', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});

describe('When form submitted', () => {
  let wrapper: Wrapper<any>;
  const localVue = createLocalVue();
  localVue.use(VueRouter);
  wrapper = mount(SearchBox, {
    localVue,
    router,
  });

  beforeEach(() => {
    wrapper.vm.$router.push('/');
  });

  it('if there is a value, then routes to search', () => {
    const query = `test query ${Date.now()}`;
    wrapper.find('input').setValue(query);
    wrapper.find('form').trigger('submit');
    expect(wrapper.vm.$route.path).toBe('/search');
    expect(wrapper.vm.$route.query.s).toBe(encodeURIComponent(query));
  });

  it('if there is no value, then routes to search', () => {
    wrapper.find('input').setValue('');
    wrapper.find('form').trigger('submit');
    expect(wrapper.vm.$route.path).not.toBe('/search');
  });
});
