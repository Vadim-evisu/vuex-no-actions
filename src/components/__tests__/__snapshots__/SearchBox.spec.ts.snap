// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox matches the snapshot when mounted 1`] = `
<div>
  <form
    action=""
  >
    <input
      class="input"
      id="searchBox"
      type="text"
    />
     
    <button
      class="button is-primary"
      id="searchSubmitButton"
    >
      Search
    </button>
  </form>
</div>
`;
